packages/selection/src/useField.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import IdManager from './utils/IdManager';
+import useControlledState from './useControlled';
+
+export default function useField(userID) {
+  const id = userID || IdManager.generateId('garden-field-container');
```

Comment:
By calling `IdManager.generateId()` directly within the hook, could this cause it to be incremented more than necessary? I'm seeing some large jumps in namespacing within Storybook.

Would it be useful to have a `useUniqueID` hook? This with a combination of `useState` hook could allow it to be in the hook "lifecycle" (or whatever they're going to call it 😄)

Reply:
I think that is a side-effect of storybook as it unmounts and mounts as you move around incrementing the internal counter, will take a closer look.

I originally has a `useID` hook that used `useState` but that then put the component into an infinite loop as each state would cause the component to re-render.

Reply:
Makes sense!